graylog2-web-interface/src/components/graylog/Panel.jsx

```diff
@@ -7,59 +9,220 @@ import teinte from 'theme/teinte';
+    }
+  }
+`;
 
 const StyledPanel = styled(BootstrapPanel)`
   background-color: ${teinte.primary.due};
 
+  .panel-group {
+    ${PanelHeading} {
+      + .panel-collapse > .panel-body,
+      + .panel-collapse > .list-group {
+        border-top-color: ${teinte.secondary.due};
+      }
+    }
+
+    ${PanelFooter} {
+      + .panel-collapse .panel-body {
+        border-bottom-color: ${teinte.secondary.due};
+      }
+    }
+  }
+
+  ${bsStyleThemeVariant(panelVariantStyles)};
+`;
+
+const deprecatedVariantStyles = hex => css`
+  /** NOTE: Deprecated & should be removed in 4.0 */
+  border-color: ${borderColor(hex)};
+
+  & > .panel-heading {
+    color: ${colorLevel(backgroundColor(hex), 9)};
+    background-color: ${backgroundColor(hex)};
+    border-color: ${borderColor(hex)};
+
+    + .panel-collapse > .panel-body {
+      border-top-color: ${borderColor(hex)};
+    }
+    .badge {
+      color: ${backgroundColor(hex)};
+      background-color: ${hex};
+    }
+  }
+
+  & > .panel-footer {
+    + .panel-collapse > .panel-body {
+      border-bottom-color: ${borderColor(hex)};
+    }
+  }
+`;
+
+const DeprecatedStyledPanel = styled(BootstrapPanel)`
+  /** NOTE: Deprecated & should be removed in 4.0 */
+  background-color: ${teinte.primary.due};
 
   .panel-footer {
     background-color: ${teinte.secondary.tre};
     border-top-color: ${teinte.secondary.due};
   }
-
   .panel-group {
     .panel-heading {
       + .panel-collapse > .panel-body,
       + .panel-collapse > .list-group {
         border-top-color: ${teinte.secondary.due};
       }
     }
-
     .panel-footer {
       + .panel-collapse .panel-body {
         border-bottom-color: ${teinte.secondary.due};
       }
     }
   }
 
-  ${bsStyleThemeVariant(panelVariantStyles)}
+  ${bsStyleThemeVariant(deprecatedVariantStyles)}
 `;
 
-export default StyledPanel;
+const Panel = ({
+  title,
+  children,
+  collapsible,
+  defaultExpanded,
+  expanded,
+  footer,
+  header,
+  onToggle,
+  ...props
+}) => {
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  React.useEffect(() => {
+    setIsExpanded((defaultExpanded && expanded)
+      || (!defaultExpanded && expanded)
+      || (defaultExpanded && isExpanded === expanded));
+  }, [expanded]);
+
+  const handleToggle = (nextIsExpanded) => {
+    setIsExpanded(nextIsExpanded);
+    onToggle(nextIsExpanded);
+  };
+
+  if (header || footer || title || collapsible || typeof children === 'string') {
+    /** NOTE: Deprecated & should be removed in 4.0 */
+    /* eslint-disable-next-line no-console */
+    console.warn('Graylog Notice: ', 'You have used a deprecated `Panel` prop, please check the documentation to use the latest `Panel`.');
+
+    return (
+      /* NOTE: this exists as a deprecated render for older Panel instances */
+      <DeprecatedStyledPanel expanded={isExpanded}
+                             onToggle={handleToggle}
+                             {...props}>
+        {(header || title) && (
```

Comment:
Kind of nitpicking, but headers look much larger now than before. I guess this is not a bug, and maybe it's even good, but just wanted to mention in case some `Panel` looks odd. Here's a before/after:

<img width="806" alt="Screenshot 2019-12-13 at 11 44 14" src="https://user-images.githubusercontent.com/716185/70794733-a6480f00-1d9e-11ea-9bcb-3cf685f89d3d.png">
<img width="716" alt="Screenshot 2019-12-13 at 11 44 22" src="https://user-images.githubusercontent.com/716185/70794737-a811d280-1d9e-11ea-8ea5-87ba702fc42c.png">